src/core/exec-invoke.c

```diff
@@ -5498,6 +5498,15 @@ int exec_invoke(
                                         return log_error_errno(errno, "Failed to disable THP per process: %m");
                                 }
                         }
+                } else if (streq(context->memory_thp, "madvise")) {
```

Comment:
ah, so this is an enum, hence please implement it like that. there are various examples for that.